src/_shared/js/dom.js

```diff
@@ -13,15 +13,34 @@ if( ! window.requestAnimationFrame ) {
     window.requestAnimationFrame = window.webkitRequestAnimationFrame || (callback => setTimeout(callback, 1000 / 60));
 }
 
-export function write(fn) {
+function write(fn, ...args) {
     return new Promise(resolve => window.requestAnimationFrame(() => {
-        fn();
-        resolve();
+        resolve(fn(...args));
     }));
 }
 
-export function read(fn) {
+function read(fn, ...args) {
     return new Promise(resolve => window.requestAnimationFrame(() => {
-        resolve(fn());
+        resolve(fn(...args));
     }));
 }
+
+function createElement(tagName) {
+    return (props = {}, attrs = {}) => {
+        let elem = document.createElement(tagName);
+        Object.assign(elem, props);
+        Object.keys(attrs).forEach(attr => elem.setAttribute(attr, attrs[attr]));
+        return elem;
+    }
+}
+
+let img = createElement('img');
+let div = createElement('div');
+
+export {
+    closest,
+    write,
+    read,
+    img,
+    div
```

Comment:
Just curious, why export these two elements? Is it because we're using them so often?